build/errors.d.ts

```diff
@@ -0,0 +1,257 @@
+/**
+ * @module errors
+ */
+import { TypedError } from 'typed-error';
+export declare class BalenaError extends TypedError {
+    code: string;
+    exitCode: number;
+}
+/**
+ * @summary Balena invalid device type
+ * @class
+ * @public
+ *
+ * @param {String} type - device type
+ * @return {Error} error instance
+ *
+ * @example
+ * throw new errors.BalenaInvalidDeviceType('raspberry-pi')
+ */
+export declare class BalenaInvalidDeviceType extends BalenaError {
+    type: string;
+    constructor(type: string);
```

Comment:
You can declare these as `constructor(public type: string);` to avoid all the duplication

Reply:
Isn't this autogenerated from the ts sources?

Reply:
It is indeed.

Reply:
Oh right this is a `.d.ts`, I didn't know we committed then

Reply:
We probably shouldn't be, it's just older projects that still do. We should fix that (here and all sorts of other places too), but I don't really want to get into it before the rename.